digital-signage-app-v2/app/components/content-area.js

```diff
@@ -0,0 +1,93 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  displayVideoSelect: true,
+  displayVideo: false,
+  focus: false,
+  video: null,
+  videoPlaying: false,
+
+  //Sends an action after the html has been rendered. Necessary to allow access
+  //to the component's wrapper div
+  didRender() {
+    this.send('updateFocus', this.get('focus'));
+  },
+  //checks input on key down to see if it is valid
+  //This needs to be pulled from the model later, not be hard coded
+  keyDown(event) {
+    if (this.get('focus')) {
+      switch (String.fromCharCode(event.keyCode).toLowerCase()) {
+        case this.get('data.config.keyboard.select'):
+          this.send('select');
+          break;
+        case this.get('data.config.keyboard.previous'):
+          //this.send('goPrevious');
+          break;
+        case this.get('data.config.keyboard.cancel'):
+          this.send('cancel');
+          break;
+        case this.get('data.config.keyboard.next'):
+          //this.send('goNext');
+          break;
+      }
+    }
+  },
+  click() {
+    this.send('updateFocus', false);
+    this.send('showVideoSelect');
+  },
+  //adds an observer for the parameter that was passed. fires when it is changed
+  //up a level. allows for change of fovus to the lists
+  changeFocusObserver: Ember.observer('focus', function() {
+    this.send('updateFocus', this.get('focus'));
+  }),
+
+  actions: {
+    select() {
+      this.set('videoPlaying', false);
+      this.send('updateFocus', false);
+      this.send('showVideoSelect');
+    },
+    cancel() {
+      this.set('videoPlaying', !this.get('videoPlaying'));
+    },
```

Comment:
Difference is between a click select and a keypress select. Parts of both can be pulled out into a new function called `updateVideo(data)` or something